code/controllers/subsystem/machinery.dm

```diff
@@ -11,11 +11,14 @@ var/list/machines = list()
 	display_order = SS_DISPLAY_MACHINERY
 
 	var/list/currentrun
+	var/currentrun_index
 
+	var/obj/machinery/M //Machine currently being processed.
```

Comment:
Accessing a datum var is slower than accessing a local var.
Also, I don't see where this reference is cleared, I don't think it is. It should get constantly overwritten by future calls of fire() because the machines list is never empty, but still. This probably isn't the way to go.